File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    st = []
    arr = list(s)
    for i in range(len(arr)):
        if st == []:
            if arr[0] == ")":
                return False
                break
            else:
                st.append("(")
        else:
            if st[-1] == "(" and arr[i] == ")":
                st.pop()
            else:
                st.append(arr[i])
    if st == []:
        answer = True
    else:
        answer = False
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        char[] arr = s.toCharArray();
        
        for(char val : arr) {
            if (!stack.isEmpty()) {
                if(stack.peek() == '(' && val == ')') {
                    stack.pop();
                    continue;
                }
            }
            stack.push(val);
        }
        return stack.isEmpty();
    }
}